addons/mail/static/src/components/thread_view/thread_view_tests.js

```diff
@@ -1181,6 +1181,146 @@ QUnit.test('Post a message containing an email address followed by a mention on
+    );
+    assert.containsNone(
+        document.body,
+        '.o_MessageList_separatorNewMessages',
+        "still no 'new messages' separator shown when current partner posted a command"
+    );
+});
+
+QUnit.test('new messages separator on receiving a message after having posted a command while composer is not focus [REQUIRE FOCUS]', async function (assert) {
+    // The goal of removing the focus is to ensure the thread is not marked as seen automatically.
+    // Indeed that would trigger channel_seen no matter what, which is already covered by other tests.
+    // The goal of this test is to cover the conditions specific to transient messages,
+    // and the conditions from focus would otherwise shadow them.
+    assert.expect(5);
+
+    // Needed partner & user to allow simulation of message reception
+    this.data['res.partner'].records.push({
+        id: 11,
+        name: "Foreigner partner",
+    });
+    this.data['res.users'].records.push({
+        id: 42,
+        name: "Foreigner user",
+        partner_id: 11,
+    });
+    this.data['mail.channel'].records = [{
+        channel_type: 'channel',
+        id: 20,
+        is_pinned: true,
+        message_unread_counter: 0,
+        seen_message_id: 10,
+        name: "General",
+        uuid: 'channel20uuid',
+    }];
+    this.data['mail.message'].records.push({
+        body: "first message",
+        channel_ids: [20],
+        id: 10,
+    });
+    await this.start();
+    const thread = this.env.models['mail.thread'].findFromIdentifyingData({
+        id: 20,
+        model: 'mail.channel'
+    });
+    const threadViewer = this.env.models['mail.thread_viewer'].create({
+        hasThreadView: true,
+        thread: [['link', thread]],
+    });
+    await this.createThreadViewComponent(threadViewer.threadView, { hasComposer: true });
+
+    assert.containsOnce(
+        document.body,
+        '.o_Message',
+        "should display one message in thread initially"
+    );
+    assert.containsNone(
+        document.body,
+        '.o_MessageList_separatorNewMessages',
+        "should not display 'new messages' separator"
+    );
+
+    document.querySelector('.o_ComposerTextInput_textarea').focus();
+    await afterNextRender(() => document.execCommand('insertText', false, "/who"));
+    await afterNextRender(() => {
+        document.querySelector('.o_Composer_buttonSend').focus();
+        document.querySelector('.o_Composer_buttonSend').click();
+    });
+    // simulate receiving a message
+    await afterNextRender(() => this.env.services.rpc({
+        route: '/mail/chat_post',
+        params: {
+            context: {
+                mockedUserId: 42,
+            },
+            uuid: 'channel20uuid',
+            message_content: "test",
+        },
+    }));
+    assert.containsN(
+        document.body,
+        '.o_Message',
+        3,
+        "should display 3 messages (initial, the transient & the received message), after posting a command"
```

Comment:
to ensure we are in a generic situation where there are messages and we received new one; not the specific case going from empty message list to non empty

Reply:
But at this step it would already be not empty due to the transient one no?

Test title (and flux that we want to test) for me does not imply there should be a message initially. Just there is at least a transient message, and then a new message coming.

For testing that the separator is just before the received message, the important distinction is that is is not before the transient message.

Does the initial message really play a role in that situation?

Reply:
In fact removing makes the test fail because of one those blurry cases when computing unread message counter.

```python
if (!this.lastSeenByCurrentPartnerMessageId) {
     return this.serverMessageUnreadCounter;
}
```

This would return 0 if user didn't see any message in the thread, whatever the situation.
So these two different situations would lead to the same UI :
- Having seen no messages when getting the unread counter from server (in messaging_init) so no new messages, no separator, great
- Same situation BUT then receive some messages, as they are not seen, `lastSeenByCurrentPartnerMessageId` is still undefined

I think we should do some work here as if there are non-transient messages in the thread not authored by current partner and if the `lastSeenByCurrentPartnerMessageId` is undefined, then the separator should be shown before the first received message which is top of the message list or after the transient message if any.

Reply:
Thanks for looking it up, explanation was clear and I agree with your conclusion.

Reply:
That means we should do the fix, keep the current test but explain why the initial message is needed, and then make a new test to cover the case with no initial message.